interp/builtin.go

```diff
@@ -553,3 +595,45 @@ func (r *Runner) relPath(path string) string {
 	}
 	return filepath.Clean(path)
 }
+
+type getopts struct {
+	argidx  int
+	runeidx int
+}
+
+func (g *getopts) Next(optstr string, args []string) (opt rune, optarg string, done bool) {
```

Comment:
Perhaps `done == (optarg == "")`? Unless `string(opt)` or `optarg` can be empty string below.

Reply:
That would be wrong in the case of an option with no arguments. But I agree this is ugly. Was thinking about a bufio.Scanner like API. getopts.Next(optstr string, args) bool and provide accessors for opt/optargs, so something like
```go
for opts.Next(optstr, args) {
       // use opts.Opt(), opts.Optarg()
}
```
would work.

Reply:
Ah, I see. I agree that something like `bufio.Scanner` would be easier to use.

However, at the moment there is a single use of this API, and it's not exported, so I'm not particularly concerned about going one way or another. I'll merge this for now, but feel free to rewrite it into the other style if you wish.